src/rgw/setup/rgw.py

```diff
@@ -394,23 +394,35 @@ def _create_rgw_user(conf: MappedConf):
             {access_key} --secret {password} --display-name="{user_name}" \
             --caps="users=*;metadata=*;usage=*;zone=*;info=*;user=*;roles=*;user-policy=*;buckets=*" \
             -c {rgw_config} -n client.radosgw-admin --no-mon-config'
-        _, err, rc, = SimpleProcess(create_usr_cmd).run(timeout=const.ADMIN_CREATION_TIMEOUT)
-        if rc == 0:
-            Log.info(f'RGW admin user {user_name} is created.')
-            return 0
-        elif rc != 0:
-            err = err.decode(const.UTF_ENCODING) if isinstance(err, bytes) else err
-            if err_str in err:
-                Log.info(f'RGW admin user {user_name} is already created,'
-                    ' Skipping user creation.')
-                return 0
-            elif timeout_str in err:
-                Log.info('RGW user creation process exceeding timeout value - '
-                    f'{const.ADMIN_CREATION_TIMEOUT} seconds. Skipping user creation on this node.')
-                return rc
+
+        # Adding retry logic for user creation with given timeout value.
+        retry_count = 0
+        return_code = -1
+        while(retry_count < const.USER_CREATION_MAX_RETRY_COUNT):
+            _, err, rc, = SimpleProcess(create_usr_cmd).run(timeout=const.ADMIN_CREATION_TIMEOUT)
+            if rc == 0:
+                Log.info(f'RGW admin user {user_name} is created.')
+                return_code = 0
             else:
-                Log.error(f'"{create_usr_cmd}" failed with error {err}.')
-                return rc
+                err = err.decode(const.UTF_ENCODING) if isinstance(err, bytes) else err
+                if err_str in err:
+                    Log.info(f'RGW admin user {user_name} is already created,'
+                        ' Skipping user creation.')
+                    return_code = 0
```

Comment:
rc = 0

Reply:
done

Reply:
this check is required because , if user is already created by another node then this rc bbecomes non zero code. but we need to return success code as 0 instead of negative rc.
hence keeping this rc assignment as it is